index.js

```diff
@@ -1,4 +1,18 @@
-/** @type {import('@typescript-eslint/utils').TSESLint.Linter.RuleLevelAndOptions} */
```

Comment:
actually would prefer to keep using the `typescript-eslint` types, but the `FlatConfig` types are not published yet, so work around this temporarily:

- https://github.com/typescript-eslint/typescript-eslint/pull/7273#issuecomment-1806851053